certbot-apache/certbot_apache/override_centos.py

```diff
@@ -47,6 +55,86 @@ def get_parser(self):
             self.aug, self.option("server_root"), self.option("vhost_root"),
             self.version, configurator=self)
 
+    def _deploy_cert(self, *args, **kwargs):
+        """
+        Override _deploy_cert in order to ensure that the Apache configuration
+        has "LoadModule ssl_module..." before parsing the VirtualHost configuration
+        that was created by Certbot
+        """
+        super(CentOSConfigurator, self)._deploy_cert(*args, **kwargs)
+        if self.version < (2, 4, 0):
+            self._deploy_loadmodule_ssl_if_needed()
+
+
+    def _deploy_loadmodule_ssl_if_needed(self):
+        """
+        Add "LoadModule ssl_module <pre-existing path>" to main httpd.conf if
+        it doesn't exist there already.
+        """
+
+        loadmods = self.parser.find_dir("LoadModule", "ssl_module", exclude=False)
+
+        correct_ifmods = []  # type: List[str]
+        loadmod_args = []  # type: List[str]
+        loadmod_paths = []  # type: List[str]
+        for m in loadmods:
+            noarg_path = m.rpartition("/")[0]
+            path_args = self.parser.get_all_args(noarg_path)
+            if loadmod_args:
+                if loadmod_args != path_args:
+                    msg = ("Certbot encountered multiple different LoadModule directives "
+                           "for LoadModule ssl_module. Please remove or comment out the "
+                           "one(s) that are not in use.")
+                    raise ConfigurationError(msg)
+            else:
+                loadmod_args = path_args
+
+            loadmod_paths.append(noarg_path)
+
+            if self.parser.not_modssl_ifmodule(noarg_path):  # pylint: disable=no-member
+                # Populate the list of known !mod_ssl.c IfModules
+                nodir_path = noarg_path.rpartition("/directive")[0]
+                correct_ifmods.append(nodir_path)
+
+            if self.parser.loc["default"] in noarg_path:
```

Comment:
nit: Can we indent this block and remove the `if self.parser.not_modssl_ifmodule(noarg_path)` call below?